CMakeLists.txt

```diff
@@ -109,18 +110,24 @@ if(CCTAG_WITH_CUDA)
     # set(CUDA_NVCC_FLAGS_RELEASE "${CUDA_NVCC_FLAGS_RELEASE};-G;-g;-DNDEBUG")
   endif()
 
-  #set(CUDA_NVCC_FLAGS         "${CUDA_NVCC_FLAGS};-gencode;arch=compute_20,code=sm_20")
-  #set(CUDA_NVCC_FLAGS         "${CUDA_NVCC_FLAGS};-gencode;arch=compute_30,code=sm_30")
-  set(CUDA_NVCC_FLAGS         "${CUDA_NVCC_FLAGS};-gencode;arch=compute_35,code=sm_35")
-  set(CUDA_NVCC_FLAGS         "${CUDA_NVCC_FLAGS};-gencode;arch=compute_50,code=sm_50")
-  set(CUDA_NVCC_FLAGS         "${CUDA_NVCC_FLAGS};-gencode;arch=compute_52,code=sm_52")
-  set(CUDA_NVCC_FLAGS         "${CUDA_NVCC_FLAGS};-gencode=arch=compute_52,code=compute_52")
-
-  if(NOT CUDA_VERSION VERSION_LESS 8.0) # cuda >= 8.0
-    set(CUDA_NVCC_FLAGS         "${CUDA_NVCC_FLAGS};-gencode=arch=compute_60,code=sm_60")
-    set(CUDA_NVCC_FLAGS         "${CUDA_NVCC_FLAGS};-gencode=arch=compute_61,code=sm_61")
-    set(CUDA_NVCC_FLAGS         "${CUDA_NVCC_FLAGS};-gencode=arch=compute_62,code=sm_62")
+  if(NOT CUDA_VERSION_MAJOR LESS 9)
```

Comment:
shouldn't this be the other way around?
if cuda < 9 u have a shorter list of CC, otherwise all of them.

Reply:
@griwodz: I agree with @simogasp, it seems to be inverted, isn't it?

Reply:
It should be:
```
if(NOT CUDA_VERSION_MAJOR LESS 9)  # cuda >= 9.0
  set(CCTAG_CUDA_CC_LIST_BASIC 35 37 50 52 53 60 61 62 70 71 72)
else()
  set(CCTAG_CUDA_CC_LIST_BASIC 35 37 50 52 53)
endif()
```

Reply:
this can help
http://arnon.dk/matching-sm-architectures-arch-and-gencode-for-various-nvidia-cards/
it has the list cuda version Vs compute

Also, building for all possible (compatible) versions maybe should be an option, while the default could be to build just for the "real" gpu on the machine. Since cmake 3.12 there is an automatic way to do that 
`CUDA_SELECT_NVCC_ARCH_FLAGS` which can autodetect the compute of the gpu
https://cmake.org/cmake/help/v3.7/module/FindCUDA.html